### ISSUE TITLE
Create QA process for release

### ISSUE THREAD

**Issue by gchi25** [GSA/px-benefit-finder#1208]
### Description

Create QA process for release.

### Acceptance Criteria

Step by step QA process for release

**Comment by gchi25** [comment]
### Module Improvements

1. https://github.com/GSA/px-benefit-finder/issues/849

http://localhost/admin/content?combine=COVID&type=bears_benefit&status=All&langcode=en
- [ ] Go to benefit "COVID-19 funeral assistance" edit page.

- [ ] Test 1: Test acceptable values field empty
- [ ] Make acceptable values field empty.
- [ ] Click "Save".

<img width="600" alt="image" src="https://github.com/GSA/px-benefit-finder/assets/88853916/d6df78f1-2b16-4527-8cb0-c9f74329737b">

You get error notifications and cannot save.

<img width="400" alt="image" src="https://github.com/GSA/px-benefit-finder/assets/88853916/7b7bca00-7a33-4110-b27a-6ea8451ebcf6">

- [x] Test 2: Test acceptable value is not an option value of criteria
Input acceptable value "Unsure".

<img width="600" alt="image" src="https://github.com/GSA/px-benefit-finder/assets/88853916/ace05cdf-0b62-4ff0-9b70-b95fac88007e">

You get error notifications and cannot save.

<img width="400" alt="image" src="https://github.com/GSA/px-benefit-finder/assets/88853916/2248c120-2f35-4c04-8679-8b3b4aa6aef7">

**Comment by gchi25** [comment]
### Module Improvements

2. https://github.com/GSA/px-benefit-finder/issues/1156

- [ ] Test 1: Can not delete an agency if still used
- [ ] navigate to http://localhost/admin/content?combine=&type=bears_agency&status=All&langcode=All
- [ ] Go to agency "Social Security Administration (SSA)" edit page
- [ ] click Delete button
- [ ] verify that it lists the benefits that still use the agency Social Security Administration (SSA)
- [ ] verify that no Delete button

![image](https://github.com/GSA/px-benefit-finder/assets/88853916/0fec0ae3-36de-491b-9af5-17d9f0636e0b)

- [ ] Test 2: Can delete a criteria if not used
- [ ] Navigate to http://localhost/node/add/bears_criteria
- [ ] Create a criteria
- [ ] Edit the newly created criteria
- [ ] Click Delete button
- [ ] Verify that no content listed as this criteria is not used in any content
- [ ] Verify that Delete button showing

![image](https://github.com/GSA/px-benefit-finder/assets/88853916/0fec0ae3-36de-491b-9af5-17d9f0636e0b)

- [ ] click Delete button
- [ ] verify that the criteria is deleted

**Comment by gchi25** [comment]
### Module Improvements

3. https://github.com/GSA/px-benefit-finder/issues/1159

- [ ] Navigate to add criteria page: http://localhost/node/add/bears_criteria
- [ ] Go to agency "Social Security Administration (SSA)" edit page
- [ ] confirm "Child dependency option" field is invisible if "has child" field not checked
<img width="200" src="https://github.com/GSA/px-benefit-finder/assets/88853916/4295a468-5c63-41a0-a68c-96432b66db9f">

- [ ] check "has child" field
- [ ] confirm "Child dependency option" field become visible and required
<img width="400" src="https://github.com/GSA/px-benefit-finder/assets/88853916/02fe844d-e895-4a9e-87aa-9e9c909b61ed">

**Comment by gchi25** [comment]
### Module Improvements

4. https://github.com/GSA/px-benefit-finder/issues/1164

- [ ] navigate to http://localhost/benefit-finder/death
- [ ] open web browser  inspector
- [ ] In sources select, `benefit-finder.min.js`
- [ ] set a break point at line 1
- [ ] refresh web page
- [ ] examine H1 from life event page template
- [ ] ensure content from node is available to DOM but not visible in UI
- [ ] step forward
- [ ] note if JS rehydrates DOM
- [ ] examine H1 from React JS
- [ ] navigate app steps and ensure only one H1 is present in DOM

https://github.com/GSA/px-benefit-finder/assets/37077057/b65d005b-9ea4-4749-84c3-4035d45d8dd5

**Comment by gchi25** [comment]
### Module Improvements

5. https://github.com/GSA/px-benefit-finder/issues/1166


- [ ] Navigate to http://localhost/admin/content?combine=&type=bears_life_event&status=All&langcode=All
- [ ] Edit life event Benefit finder: death of a loved one 
- [ ] Add meta description "Find federal benefits fast. Answer a few questions and in a few minutes get your list of possible benefits across different agencies."
- [ ] Save
- [ ] Navigate to http://localhost/benefit-finder/death
- [ ] Examine meta description in header

![image](https://github.com/GSA/px-benefit-finder/assets/88853916/510f8579-c4a3-45af-8df8-f49ee0bd866e)

**Comment by gchi25** [comment]
### Module Improvements

6. https://github.com/GSA/px-benefit-finder/issues/1171


- [ ] Navigate to http://localhost/admin/content?combine=&type=bears_life_event&status=All&langcode=All
- [ ] Edit life event Benefit finder: death of a loved one 
- [ ] Add meta description "Find federal benefits fast. Answer a few questions and in a few minutes get your list of possible benefits across different agencies."
- [ ] Save
- [ ] Navigate to http://localhost/benefit-finder/death
- [ ] Examine meta description in header

![image](https://github.com/GSA/px-benefit-finder/assets/88853916/510f8579-c4a3-45af-8df8-f49ee0bd866e)

**Comment by gchi25** [comment]
### Module Improvements

7. https://github.com/GSA/px-benefit-finder/issues/1226

- [ ] Navigate to http://localhost/admin/content?combine=&type=bears_life_event_form&status=All&langcode=All
- [ ] Go to life event form "Benefit finder: death of a loved one" edit page
- [ ] Save as draft
- [ ] Verify that it shows progress bar with title "Generating only DRAFT JSON data files"

<img width="600" src="https://github.com/GSA/px-benefit-finder/assets/88853916/c8063f1d-67b6-4d4b-bf4c-ae7871d5405b">

- [ ] Edit the life event form again
- [ ] Save as published
- [ ] Verify that it shows progress bar with title "Generating both DRAFT and REAL JSON data files"

<img width="600" src="https://github.com/GSA/px-benefit-finder/assets/88853916/394b10a7-c234-4a8e-9330-478817bac0fd">

**Comment by gchi25** [comment]
### App Improvements

2. https://github.com/GSA/px-benefit-finder/issues/1182

- [ ] Navigate to http://localhost/benefit-finder/death
- [ ] Confirm that notices list renders in the initial view with the `info` icon

![Screenshot 2024-04-12 111310](https://github.com/GSA/px-benefit-finder/assets/37077057/8c78540f-ef43-4f4e-ade8-2b2e9552c1d7)

**Comment by gchi25** [comment]
### App Improvements

3. https://github.com/GSA/px-benefit-finder/issues/1073

- [ ] Navigate to results view
- [ ] Toggle on axeDev tool
- [ ] Run scan
- [ ] Confirm aria error is resolved
- [ ] Inspect elements
- [ ] Confirm that only the `<button />` elements in the accordion components have `aria-extended` attributes

expected:
<img width="1437" alt="Screenshot 2024-04-11 at 10 45 10 AM" src="https://github.com/GSA/px-benefit-finder/assets/37077057/4cdb70aa-a19d-4881-9d21-7cce7dd0c052">

**Comment by gchi25** [comment]
### App Improvements

4. https://github.com/GSA/px-benefit-finder/issues/1183

- [ ] Navigate to initial questions
- [ ] Confirm default state of error alert is aria-hidden="true"
<img width="600" src="https://github.com/GSA/px-benefit-finder/assets/88853916/89a36d18-d80c-49bf-8741-934ebae25e9c">

- [ ] Trigger error by moving forward without completing required fields
- [ ] Confirm default state updated to aria-hidden="false"
<img width="600" src="https://github.com/GSA/px-benefit-finder/assets/88853916/e99add2e-82d6-4a7a-8696-c94d58e8d002">

**Comment by gchi25** [comment]
### App Improvements

5. https://github.com/GSA/px-benefit-finder/issues/1180

- [ ] review code, ensure href has new attribute and values
<img width="600" src="https://github.com/GSA/px-benefit-finder/assets/88853916/9e593a82-ff99-4430-aa61-da7e7ba1ec10">

**Comment by gchi25** [comment]
### App Improvements

6. https://github.com/GSA/px-benefit-finder/issues/1072

- [ ] Navigate to step one of the form
- [ ] Trigger an error by clicking continue
- [ ] Inspect DOM
- [ ] Ensure that heading levels in errors are <h2> while remaining consistent in style to production
<img width="100%" src="https://github.com/GSA/px-benefit-finder/assets/88853916/2a344e32-7a6e-4c29-886d-ed68183852b1">

**Comment by gchi25** [comment]
### App Improvements

7. https://github.com/GSA/px-benefit-finder/issues/1119

- [ ] Navigate to modal step
- [ ] Inspect
- [ ] Ensure that heading in modal is `<h1 />`
- [ ] Ensure that styles did not change
<img width="100%" src="https://github.com/GSA/px-benefit-finder/assets/88853916/a5f2015b-dd87-45ac-ad58-9d84ce299d2d">

**Comment by scottqueen-bixal** [comment]
This is such a beautiful ticket 😍

**Comment by diegocob** [comment]
Thanks for walkimg us step by step on the QA, @gchi25. You are the best😻. Lots of detail and care in this teicket.

Ran through almost all content QA steps. A few issues: 
1. Point #4 couldn't complete. Wasn't able to select in sources, benefit-finder.min.js and set a break point at line 1 
2. Couldn't run the a11y tickets with axedev tools On app improvments: 
- couldn't run a11y test
- couldn't run https://github.com/GSA/px-benefit-finder/issues/1073

Quick notes:
1. When I deleted a content node on the dev cms it would take me out to the main bf-cms-dev.bxdev.net page
![image](https://github.com/GSA/px-benefit-finder/assets/125094772/3b000667-f45e-4a46-a7f0-2f1b434b99ce)
2. Whenever I created a new node it would take time to generate the json files. This didn't happen before, not sure if it's normal.
![image](https://github.com/GSA/px-benefit-finder/assets/125094772/1af2568a-da25-4051-92e5-187502e63766)
3. Later I'd like to learn more about the child dependency on the criteria and why it's there.
